### PR TITLE
12-setup-react-router

### PR DESCRIPTION
**Pull request by evanhameed99**
I basically set up react router just as the issue clarifies and demands.
Yet I have a tweak that we can consider before closing this issue. Now we all know, except the landing page, all other pages will basically have a Navbar and Footer components. With that being said, I am not pretty sure if we can achieve this scenario with the way this issue demands the code to be. 
We can talk more about that in our daily stand-up today!

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
+import Meals from './containers/Meals/Meals';
+import Workouts from './containers/Workouts/Workouts';
+import About from './containers/About/About';
+import Blogs from './containers/Blogs/Blogs';
 function App() {
   return (
-    <div>
-      <h1 className="text-3xl text-purple-500">Train To Maintain</h1>
-    </div>
+    <React.Fragment>
```
Comment (osamaakb) [inline src/App.js:11]: you can change this to 
```suggestion
    <>
```

**Comment by osamaakb** [comment]
I would love to see less commits too, commit messages are usually for an entire feature or a fix or style change, so for example this pull request could of been under one commit called `Feat: add react router` and that's it, try to work on that more

**Comment by evanhameed99** [comment]
I already saw and checked this question and it was really helpful. I was planning to show you something similar.
Anyways will do it soon. 
Thank you

File: src/App.js [hunk 2]
```diff
@@ -1,10 +1,43 @@
...
 
 function App() {
   return (
-    <div>
-      <h1 className="text-3xl text-purple-500">Train To Maintain</h1>
-    </div>
+    <>
+      <Router>
+        <Switch>
+          <Route component={NotFound} path="/notfound" />
```
Comment (osamaakb) [inline src/App.js:35]: the not found or 404 shouldn't have a specific path but instead be the last `Router`

File: src/App.js [hunk 3]
```diff
@@ -1,10 +1,43 @@
...
+          <Route exact path="/" component={LandingPage} />
+          <Route component={DefaultRoutes} />
```
Comment (osamaakb) [inline src/App.js:37]: so it should be under this line 
```jsx 
<Route component={NotFound} />
```

File: src/App.js (hunk 2 above)
Comment (evanhameed99) [reply src/App.js:35]: To be honest I searched a lot about that and yes in normal cases the 404 page should the last route and also shouldn't have a path. In our case, we are trying to make  nested routes to make the navbar and the footer sticks in certain pages and also redirect to 404 page in the main route. I tried to implement it the way you suggested by putting it at the bottom without a path, yet I think it is not possible in our case. 
This is the reference I am relying on [codesandbox](https://codesandbox.io/s/cool-chaplygin-5to6g?file=/src/index.js )

File: src/containers/NotFound/RedirectToNotFound.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+const RedirectToNotFound = () => {
```
Comment (osamaakb) [inline src/containers/NotFound/RedirectToNotFound.js:3]: No need for this

File: src/App.js [hunk 5]
```diff
@@ -1,10 +1,43 @@
...
+  return (
+    <>
+      <Navbar />
+      <Switch>
+        <Route path="/home" component={Home} />
+        <Route path="/meals" component={Meals} />
+        <Route path="/workouts" component={Workouts} />
+        <Route path="/about" component={About} />
+        <Route path="/blogs" component={Blogs} />
+        <Route component={RedirectToNotFound} />
```
Comment (osamaakb) [inline src/App.js:23]: replace this with this
```suggestion
        <Redirect to="notfound" />
```

**Pull request merged**